OpenEdX/Router.swift

```diff
@@ -854,4 +866,112 @@ extension Router {
+        }
+    }
+}
+
+// MARK: Payments
+extension Router {
+    @MainActor
+    public func showUpgradeInfo(
+        productName: String,
+        message: String,
+        sku: String,
+        courseID: String,
+        screen: CourseUpgradeScreen,
+        pacing: String
+    ) async {
+        await withCheckedContinuation { continuation in
+            let view = UpgradeInfoSheetView(
+                viewModel: Container.shared.resolve(
+                    UpgradeInfoViewModel.self,
+                    arguments: productName, message, sku, courseID, screen, pacing
+                )!
+            )
+            let controller = UIHostingController(rootView: view)
+            if let sheet = controller.sheetPresentationController {
+                sheet.detents = [.large()]
+                sheet.prefersEdgeAttachedInCompactHeight = true
+                sheet.widthFollowsPreferredContentSizeWhenEdgeAttached = true
+                sheet.prefersGrabberVisible = true
+            }
+            navigationController.present(controller, animated: true) {
+                continuation.resume()
+            }
+        }
+    }
+    
+    @MainActor
+    public func hideUpgradeInfo(animated: Bool) async {
+        await withCheckedContinuation { continuation in
+            if let controller = navigationController.presentedViewController as? UIHostingController<UpgradeInfoSheetView> {
+                controller.dismiss(animated: animated) {
+                    continuation.resume()
+                }
+            } else {
+                continuation.resume()
+            }
+        }
+    }
+    
+    @MainActor
+    public func showUpgradeLoaderView(animated: Bool) async {
+        await withCheckedContinuation { continuation in
+            let unlockView = CourseUpgradeUnlockView()
+            let controller = UIHostingController(rootView: unlockView)
+            controller.modalTransitionStyle = .crossDissolve
+            controller.modalPresentationStyle = .overFullScreen
+            navigationController.present(controller, animated: animated) {
+                continuation.resume()
+            }
+        }
+    }
+    
+    @MainActor
+    public func hideUpgradeLoaderView(animated: Bool) async {
+        await withCheckedContinuation { continuation in
+            let presentedController = navigationController.presentedViewController
+            if let controller = presentedController as? UIHostingController<CourseUpgradeUnlockView> {
+                controller.dismiss(animated: animated) {
+                    continuation.resume()
+                }
+            } else {
+                continuation.resume()
+            }
+        }
+    }
+    
+    @MainActor
+    public func showRestoreProgressView() {
+            let unlockView = RestoreInProgressView()
+            let controller = UIHostingController(rootView: unlockView)
+
+        controller.view.frame = CGRect(x: 0,
```

Comment:
please move each parameter on own line

Reply:
This is how it looks if moved all the parameters. It's loosing the readability instead if improving it. Thats why kept first parameter on the same line of opening bracket.  
```
        controller.view.frame = CGRect(
            x: 0,
            y: 0,
            width: UIScreen.main.bounds.width,
            height: UIScreen.main.bounds.height
        )
 ```

Reply:
For my view it's good readable.